wiki.py

```diff
@@ -658,15 +658,76 @@ def single_project_info(self, project_id, sv_name):
             elif v.get("title"):
                 pages.append(self._make_year_title(v["title"]))
         logging.warning(
-            "Don't forget to manually add '{id} - {name}' to the following "
-            "pages: {pages}".format(
-                id=project_id, name=sv_name, pages='\n* '.join(pages)
+            "Don't forget to manually add '{number} - {name}' to the "
+            "following pages: {pages}".format(
+                number=number, name=sv_name, pages='\n* '.join(pages)
             )
         )
 
     def log_report(self):
-        """Log a list of the pages that were modified.
-        """
+        """Log a list of the pages that were modified."""
         logging.info("These pages were modified:")
         for page in self._touched_pages:
             logging.info(page.title())
+
+    def update_project_name_template(self):
+        """Update project number and name templates."""
+        name_template = Page(self._site, self._config["project_name_template"])
+        number_template = Page(
+            self._site,
+            self._config["project_number_template"]
+        )
+        for number, name in self._projects.items():
+            english_name = name["en"]
+            swedish_name = name["sv"]
+
+            name_row = (
+                f"| {number} = "
+                "{{#if: {{{en|}}}"
+                f"| {english_name} | {swedish_name} "
+                "}}"
+            )
+            self._insert_row_before_default(name_template, name_row, number)
```

Comment:
Do you really need to call this for each row? Wouldn't it be easier to construct a list of rows to be added and then add them all in one go? Right now this finds the default row for every project.

To handle the de-duplication when a list of rows is passed one could simply also find the row containing `#switch` then make a set of all lines between that and default and subtract this set from the incoming list.

Reply:
Maybe. Performance isn't really an issue when doing 10-20 searches, at least not the way the script is run.

Reply:
that is also very true =)